nixos/modules/services/networking/wpa_supplicant.nix

```diff
@@ -94,7 +94,7 @@ let
         ++ optional (pskString != null) "psk=${pskString}"
         ++ optionals (opts.auth != null) (filter (x: x != "") (splitString "\n" opts.auth))
         ++ optional (opts.priority != null) "priority=${toString opts.priority}"
-        ++ optional (opts.extraConfig != "") opts.extraConfig;
+        ++ optionals (opts.extraConfig != "") (filter (x: x != "") (splitString "\n" opts.extraConfig));
     in
     ''
       network={
```

Comment:
Isn't the indentation of the first line (`ssid=...`) is wrong?

Reply:
No, it indents correctly